templates/clusterclass-template.yaml

```diff
@@ -0,0 +1,611 @@
+        type: integer
+  - name: controlPlaneImageID
+    required: true
+    schema:
+      openAPIV3Schema:
+        description: ImageID for the control plane machines.
+        type: string
+  - name: workerNumCores
+    required: true
+    schema:
+      openAPIV3Schema:
+        description: NumCores for the worker machines.
+        type: integer
+  - name: workerMemoryMB
+    required: true
+    schema:
+      openAPIV3Schema:
+        description: MemoryMB for the worker machines.
+        type: integer
+  - name: workerImageID
+    required: true
+    schema:
+      openAPIV3Schema:
+        description: ImageID for the worker machines.
+        type: string
+  - name: credsSecretName
+    required: true
+    schema:
+      openAPIV3Schema:
+        description: Secret containing the credentials for the infra cluster.
+        type: string
+  - name: location
+    required: true
+    schema:
+      openAPIV3Schema:
+        description: cluster location on IONOS
+        type: string
+  workers:
+    machineDeployments:
+    - class: worker
+      template:
+        bootstrap:
+          ref:
+            apiVersion: bootstrap.cluster.x-k8s.io/v1beta1
+            kind: KubeadmConfigTemplate
+            name: ${CLUSTER_CLASS_NAME}-worker
+            namespace: '${NAMESPACE}'
+        infrastructure:
+          ref:
+            apiVersion: infrastructure.cluster.x-k8s.io/v1alpha1
+            kind: IonosCloudMachineTemplate
+            name: ${CLUSTER_CLASS_NAME}-worker-machinetemplate
+            namespace: '${NAMESPACE}'
+        metadata: {}
+---
+kind: KubeadmControlPlaneTemplate
+apiVersion: controlplane.cluster.x-k8s.io/v1beta1
+metadata:
+  name: "${CLUSTER_CLASS_NAME}-controlplane"
+  namespace: '${NAMESPACE}'
+spec:
+  template:
+    spec:
+      kubeadmConfigSpec:
+        ntp:
+            enabled: true
+            servers:
+              - 0.de.pool.ntp.org
+              - 1.de.pool.ntp.org
+              - 2.de.pool.ntp.org
+              - 3.de.pool.ntp.org
+        files:
+          - path: /etc/ssh/sshd_config.d/ssh-audit_hardening.conf
+            owner: root:root
+            permissions: '0644'
+            content: |
+              # Restrict key exchange, cipher, and MAC algorithms, as per sshaudit.com
+              # hardening guide.
+              KexAlgorithms curve25519-sha256,curve25519-sha256@libssh.org,diffie-hellman-group16-sha512,diffie-hellman-group18-sha512,diffie-hellman-group-exchange-sha256
+              Ciphers chacha20-poly1305@openssh.com,aes256-gcm@openssh.com,aes128-gcm@openssh.com,aes256-ctr,aes192-ctr,aes128-ctr
+              MACs hmac-sha2-256-etm@openssh.com,hmac-sha2-512-etm@openssh.com,umac-128-etm@openssh.com
+              HostKeyAlgorithms ssh-ed25519,ssh-ed25519-cert-v01@openssh.com,sk-ssh-ed25519@openssh.com,sk-ssh-ed25519-cert-v01@openssh.com,rsa-sha2-256,rsa-sha2-512,rsa-sha2-256-cert-v01@openssh.com,rsa-sha2-512-cert-v01@openssh.com
```

Comment:
https://github.com/ionos-cloud/cluster-api-provider-ionoscloud/pull/140 will soon be merged. Please sync the ssh config from main then.